app/src/main/java/org/simple/clinic/login/pin/LoginPinUpdate.kt

```diff
@@ -1,11 +1,25 @@
 package org.simple.clinic.login.pin
 
 import com.spotify.mobius.Next
-import com.spotify.mobius.Next.noChange
 import com.spotify.mobius.Update
+import org.simple.clinic.mobius.dispatch
+import org.simple.clinic.mobius.next
 
 class LoginPinUpdate : Update<LoginPinModel, LoginPinEvent, LoginPinEffect> {
   override fun update(model: LoginPinModel, event: LoginPinEvent): Next<LoginPinModel, LoginPinEffect> {
-    return noChange()
+    return when (event) {
+      is OngoingLoginEntryLoaded -> next(
+          model.ongoingLoginEntryUpdated(event.ongoingLoginEntry)
+      )
+      is LoginPinAuthenticated -> next(
+          model.ongoingLoginEntryUpdated(event.newLoginEntry),
+          SaveOngoingLoginEntry(event.newLoginEntry)
+      )
+      is LoginPinScreenUpdatedLoginEntry -> next(
+          model.ongoingLoginEntryUpdated(event.ongoingLoginEntry),
```

Comment:
Is it necessary to emit a model change here since the entry is not really changing at all and it is already present in the model?

Reply:
Yeah, I wasn't sure about it either but kept it to make sure I haven't missed anything during migration. Will change it to just dispatch the effect.

Reply:
Done